app/controllers/schools/transferring_participants_controller.rb

```diff
@@ -123,46 +123,18 @@ def cannot_add; end
     #   b. Send email to participant to notify them.
     #
     def send_notification_emails!(induction_record, was_withdrawn_participant)
-      current_lead_provider_users = current_lead_provider&.users || []
-      target_lead_provider_users = participant_lead_provider&.users || []
-
-      if was_withdrawn_participant
-        current_lead_provider_users.each do |user|
-          ParticipantTransferMailer.provider_transfer_in_notification(
-            induction_record:,
-            lead_provider_profile: user.lead_provider_profile,
-          ).deliver_later
-        end
-      elsif matching_lead_provider_and_delivery_partner?
-        current_lead_provider_users.each do |user|
-          ParticipantTransferMailer.provider_existing_school_transfer_notification(
-            induction_record:,
-            lead_provider_profile: user.lead_provider_profile,
-          ).deliver_later
-        end
-      elsif @transferring_participant_form.switch_to_schools_programme?
-        current_lead_provider_users.each do |user|
-          ParticipantTransferMailer.provider_transfer_in_notification(
-            induction_record:,
-            lead_provider_profile: user.lead_provider_profile,
-          ).deliver_later
-        end
-        target_lead_provider_users.each do |user|
-          ParticipantTransferMailer.provider_transfer_out_notification(
-            induction_record:,
-            lead_provider_profile: user.lead_provider_profile,
-          ).deliver_later
-        end
-      else
-        target_lead_provider_users.each do |user|
-          ParticipantTransferMailer.provider_new_school_transfer_notification(
-            induction_record:,
-            lead_provider_profile: user.lead_provider_profile,
-          ).deliver_later
-        end
-      end
-
-      ParticipantTransferMailer.participant_transfer_in_notification(induction_record:).deliver_later
+      current_lead_provider_profiles = current_lead_provider&.users&.map(&:lead_provider_profile) || []
+      target_lead_provider_profiles = participant_lead_provider&.users&.map(&:lead_provider_profile) || []
```

Comment:
Not especially fond of chaining like this, feels like this could be achieved through relationships and `through:`